controllers/storagecluster/cephobjectstores_test.go

```diff
@@ -48,7 +51,8 @@ func assertCephObjectStores(t *testing.T, reconciler StorageClusterReconciler, c
 	err = reconciler.Client.Get(context.TODO(), request.NamespacedName, actualCos)
 	// for any cloud platform, 'cephobjectstore' should not be created
 	// 'Get' should have thrown an error
-	if avoidObjectStore(reconciler.platform.platform) {
+	isavoidObjectStore, _, _ := reconciler.avoidObjectStore(cr)
```

Comment:
I see you're mostly interested in checking the err from the Get() call above. What happens if `avoidObjectStore()` fails with an error? This means that we can't trust the value of `isavoidObjectStore` because a value of `false` may not actually be correct, it's just the default. You should check the error returned from `avoidObjectStore()` separately, and return it in all cases if not nil.